components/filter/filter.js

```diff
@@ -213,11 +214,13 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		}
 		return this._dimensions.map((dimension) => {
 			const builtDimension = this._buildDimension(dimension);
-			const dimensionDescription = `${dimension.text}. ${this.localize('components.filter.filterCountDescription', { number: dimension.appliedCount })}`;
-			return html`<d2l-menu-item text="${dimension.text}" description="${dimensionDescription}">
+			const dimensionDescription = `${dimension.text}.`;
+			const countBadgeId = getUniqueId();
```

Comment:
Didn't know about this handy feature. All the Ids here need to be unique so the aria-describedby tag can read from the correct badge.

Reply:
I don't know the answer to this, maybe @dlockhart... do you know if there's a perf hit with changing this id every re-render? We could also make an id out of a combo of the dimension kay and the value key, but not sure if it matters.

Reply:
I did have the combo of the dimension and value key as my previous iteration. I was just worried it would get really messy turning the text into an id, considering it could contain different punctuation, spaces, that would have to be removed.

Reply:
That's a good point Stacey... yeah this is going to cause a perf issue in the sense that Lit is going to have to update DOM on every re-render, whereas normally if the output doesn't change they don't do anything.

You could either combine dimension key + value key (which should both be numbers or something similar) or generate a unique ID but store it in the map so that you're not re-generating it each time.

Reply:
I guess value doesn't make sense here. If you do want to use the dimension for the id, I'd suggest something like:
```
id="${MENU_ITEM_ID_PREFIX}${dimension.key}"
```
because IDs can't start with a number, and the consumer key might.
Similar to here: https://github.com/BrightspaceUI/core/blob/main/components/filter/filter.js#L316